### PR TITLE
Miniature fire extinguisher

### PR DESCRIPTION
**Pull request by JoeHammad1844**
## About the PR

Adds a miniature fire extinguisher which can fit in tool belts, contains half the water as a regular one and does 6 blunt damage. fire lockers are now guaranteed to contain a fire extinguisher, emergency lockers may contain a mini fire extinguisher, emergency toolboxes are guaranteed to contain a miniature fire extinguisher.
they also spawn in engineering belts
## Media

fire_extinguisher_closed
fire_extinguisher_open

- [x] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

**Changelog**
Make sure to take this Changelog template out of the comment block in order for it to show up.
:cl: JoeHammad
- add: Miniature fire extinguishers have been made available in youtools, emergency toolboxes and toolbelts.
- tweak: Fire extinguishers are now guaranteed to spawn in fire lockers.

### CODE REVIEW

**Comment by JoeHammad1844** [comment]
my scholars shall decipher these runes 
image

**Comment by JoeHammad1844** [comment]
yeah I dont know what this means, ill draft it until someone tells me what to do here because I was banned off the discord :trollface:

**Comment by Holinka4ever** [comment]
My favorite item
Pocket jetpack

I wish we could use it for extinguishing own self, if we are on fire/pie mark

**Comment by JoeHammad1844** [comment]
> I wish we could use it for extinguishing own self, if we are on fire/pie mark

you can just splash yourself with a cup of water

**Comment by Holinka4ever** [comment]
> > I wish we could use it for extinguishing own self, if we are on fire/pie mark
> 
> you can just splash yourself with a cup of water

Thats doesnt look ethical, cause even having a compact extinguisher, thats weird!
We cant fire off from own self, we just lay on a ground or hit by water from cup, or someone else with another fire extinguisher.

**Comment by Tunguso4ka** [comment]
> my scholars shall decipher these runes image

try adding
`
- type: Tag
  tags: 
  - FireExtinguisherMini
`
to fireextinguishermini item.

**Comment by JoeHammad1844** [comment]
can you make it a review so its easier?

**Review by lzk228 — COMMENTED** [review]
sprites looking weird, entire red inhand and white in icon

i really don't think that it should be in belts, maybe put in engivend or youtool

File: Resources/Prototypes/Entities/Objects/Misc/fire_extinguisher_mini.yml [hunk 1]
```diff
@@ -0,0 +1,51 @@
+- type: entity
+  name: miniature fire extinguisher
+  parent: BaseItem
+  id: FireExtinguisherMini
```
[the remark this answers is not in the document]
Comment (JoeHammad1844) [reply Resources/Prototypes/Entities/Objects/Misc/fire_extinguisher_mini.yml:4]: SIR YES SIR OORAH

File: Resources/Prototypes/Entities/Objects/Misc/fire_extinguisher.yml [hunk 2]
```diff
@@ -49,6 +49,57 @@
...
+  id: FireExtinguisherMini
+  description: It extinguishes fires.
+  components:
+  - type: Sprite
+    sprite: Objects/Misc/fire_extinguisher_mini.rsi
+    layers:
+        - state: fire_extinguisher_closed
+          map: [ "enabled" ]
+  - type: Item
+    sprite: Objects/Misc/fire_extinguisher.rsi
```
Comment (lzk228) [inline Resources/Prototypes/Entities/Objects/Misc/fire_extinguisher.yml:64]:
```suggestion
    sprite: Objects/Misc/fire_extinguisher_mini.rsi
```

File: Resources/Prototypes/Entities/Objects/Misc/fire_extinguisher.yml [hunk 3]
```diff
@@ -49,6 +49,57 @@
...
+        maxVol: 50
+        reagents:
+        - ReagentId: Water
+          Quantity: 50
+  - type: RefillableSolution
+    solution: spray
+  - type: DrainableSolution
+    solution: spray
+  - type: SolutionTransfer
+  - type: ItemCooldown
```
Comment (lzk228) [inline Resources/Prototypes/Entities/Objects/Misc/fire_extinguisher.yml:78]:
```suggestion
```

File: Resources/Prototypes/Entities/Objects/Misc/fire_extinguisher.yml [hunk 4]
```diff
@@ -49,6 +49,57 @@
...
+  - type: Spray
+    transferAmount: 10
```
Comment (lzk228) [inline Resources/Prototypes/Entities/Objects/Misc/fire_extinguisher.yml:80]:
```suggestion
    transferAmount: 5
```

File: Resources/Prototypes/Entities/Objects/Misc/fire_extinguisher.yml [hunk 5]
```diff
@@ -49,6 +49,57 @@
...
+    spraySound:
+      path: /Audio/Effects/extinguish.ogg
+    sprayedPrototype: ExtinguisherSpray
+    vaporAmount: 3
+    vaporSpread: 90
+    sprayVelocity: 2.0
+  - type: FireExtinguisher
+    hasSafety: true
```
Comment (lzk228) [inline Resources/Prototypes/Entities/Objects/Misc/fire_extinguisher.yml:88]:
```suggestion
```

File: Resources/Prototypes/Entities/Objects/Misc/fire_extinguisher.yml [hunk 6]
```diff
@@ -49,6 +49,57 @@
...
+        Blunt: 6
+    soundHit:
+      path: /Audio/Weapons/smash.ogg
+  - type: Appearance
+  - type: GenericVisualizer
+    visuals:
+      enum.FireExtinguisherVisuals.Safety:
+        enabled:
+          True: { state: fire_extinguisher_closed }
+          False: { state: fire_extinguisher_open }
```
Comment (lzk228) [inline Resources/Prototypes/Entities/Objects/Misc/fire_extinguisher.yml:101]:
```suggestion
```

File: Resources/Prototypes/Entities/Clothing/Belt/belts.yml [hunk 7]
```diff
@@ -29,6 +29,7 @@
         - HolofanProjector
         - Multitool
         - AppraisalTool
+        - FireExtinguisherMini
```
Comment (lzk228) [inline Resources/Prototypes/Entities/Clothing/Belt/belts.yml:32]: whitelist requires tag, not an entity id, so you should make a FireExtinguisherMini tag

**Comment by Partmedia** [comment]
Regardless of the bugs in the submission, I don't think a mini fire extinguisher is necessary or well balanced. Fire extinguishers are nominally for putting out fires, but players typically fill them up with other nasty chemicals to spray at others. As such I don't think having a mini concealable version of the fire extinguisher is good for the game.

**Comment by CatMagic1** [comment]
1. Rename to emergency fire extinguisher
2. Make it so that it can not be filled up again after it is emptied or can only be filled with water


-_-

**Comment by JoeHammad1844** [comment]
> Regardless of the bugs in the submission, I don't think a mini fire extinguisher is necessary or well balanced. Fire extinguishers are nominally for putting out fires, but players typically fill them up with other nasty chemicals to spray at others. As such I don't think having a mini concealable version of the fire extinguisher is good for the game.

the point is to have fire extinguishers more available to actually put out fires, and the only thing you can really put into a fire extinguisher to make it hurt people is clf3, which requires chemistry and to blow yourself up

File: Resources/Prototypes/Entities/Clothing/Belt/belts.yml (hunk 7 above)
Comment (JoeHammad1844) [reply Resources/Prototypes/Entities/Clothing/Belt/belts.yml:32]: so that would require it being in the fire_extinguisher.yml?

**Comment by JoeHammad1844** [comment]
i also dont see whats wrong with giving antagonists more options to be creative

File: Resources/Prototypes/Entities/Clothing/Belt/belts.yml (hunk 7 above)
Comment (JoeHammad1844) [reply Resources/Prototypes/Entities/Clothing/Belt/belts.yml:32]: in the fire_extinguisher.yml?

**Comment by Partmedia** [comment]
> the point is to have fire extinguishers more available to actually put out fires, and the only thing you can really put into a fire extinguisher to make it hurt people is clf3, which requires chemistry and to blow yourself up

You can, of course, find the nearest fire extinguisher cabinet, pull it out, and use it to put out the fire, without having to have a mini portable version of it in your belt at all times.

I've opted to close this for now. I don't feel that these reasons warrant having a separate mini belt item of a full-sized tool. If you'd like to continue discussion or have significant new arguments in favor, please feel free to re-open.

**Comment by Carolyn3114** [comment]
> > the point is to have fire extinguishers more available to actually put out fires, and the only thing you can really put into a fire extinguisher to make it hurt people is clf3, which requires chemistry and to blow yourself up
> 
> You can, of course, find the nearest fire extinguisher cabinet, pull it out, and use it to put out the fire, without having to have a mini portable version of it in your belt at all times.
> 
> I've opted to close this for now. I don't feel that these reasons warrant having a separate mini belt item of a full-sized tool. If you'd like to continue discussion or have significant new arguments in favor, please feel free to re-open.

when atmos properly interacts with fire extinguishers, these would be amazing

**Comment by JoeHammad1844** [comment]
> You can, of course, find the nearest fire extinguisher cabinet, pull it out, and use it to put out the fire, without having to have a mini portable version of it in your belt at all times.

the thing is that by the time you go and get the fire extinguisher they fire victim has either put it out by stopping, dropping or rolling, run off, or is crit. and depending on the map the extinguisher wall closets wont even be close by and fire closets arent even guaranteed to have a fire extinguisher

**Pull request closed without merging**